### PR TITLE
Fix filtering jobs

### PR DESCRIPTION
**Pull request by icanberk**

### CODE REVIEW

File: app/controllers/jobs_controller.rb [hunk 1]
```diff
@@ -1,14 +1,28 @@
- $:.unshift("/usr/share/tango2/thrift/gen-rb/")
-
 require 'autoConfig'
-require "ModuleBase.rb"
 
 class JobsController < ApplicationController
-  include ModuleBase
+  # include ModuleBase
```
Comment (dlbucci) [inline app/controllers/jobs_controller.rb]: Do we want dead code?

File: app/views/jobs/index.html.erb [hunk 2]
```diff
@@ -8,7 +8,7 @@ This page shows you the status of current and recent autograding jobs.
 
 <h2>Currently Running Jobs</h2>
 
-<% if @running_jobs == [] then %>
+<% if !@running_jobs or @running_jobs == [] then %>
```
Comment (dlbucci) [inline app/views/jobs/index.html.erb]: I think @running_jobs.blank? would be better:
http://stackoverflow.com/questions/885414/a-concise-explanation-of-nil-v-empty-v-blank-in-ruby-on-rails

File: app/views/jobs/index.html.erb [hunk 3]
```diff
@@ -37,7 +37,7 @@ This page shows you the status of current and recent autograding jobs.
 
 <h2>Jobs Waiting to Run</h2>
 
-<% if @waiting_jobs == [] then %>
+<% if !@waiting_jobs or @waiting_jobs == [] then %>
```
Comment (dlbucci) [inline app/views/jobs/index.html.erb]: .blank?

File: app/views/jobs/index.html.erb [hunk 4]
```diff
@@ -81,7 +82,7 @@ end
   <h2>Your Recently Completed Jobs <%=since%></h2>
 <% end %>
 
-<% if @dead_jobs == [] then %>
+<% if !@dead_jobs_view or @dead_jobs == [] then %>
```
Comment (dlbucci) [inline app/views/jobs/index.html.erb]: .blank?  Actually, couldn't this be an else?

File: app/controllers/jobs_controller.rb [hunk 5]
```diff
@@ -159,17 +183,24 @@ def formatRawJob(rjob, is_live)
...
-      if !@cud.instructor? then 
+      if !@cud.instructor? then
         # Students can see only their own job names
         if !job[:name][@cud.user.email] then
           job[:name] = "*"
         end
       else
         # Instructors can see only their course's job names
-        if !rjob["notifyURL"] then
+        if !rjob["notifyURL"] or !(job[:course].eql? @cud.course.id.to_s) then
```
Comment (dlbucci) [inline app/controllers/jobs_controller.rb]: Is job[:course] the numeric id of the course? Or is it the name of the course?

Comment (icanberk) [inline app/controllers/jobs_controller.rb]: job[:course] is a string here since we're just getting it by regexping the notifyUrl.

Comment (dlbucci) [inline app/controllers/jobs_controller.rb]: But is it a '15213-s15' string id or a '1234' database id?

Comment (icanberk) [inline app/controllers/jobs_controller.rb]: It's database id

**Pull request merged**